examples/with-firebase-hosting-and-docker/README.md

```diff
@@ -0,0 +1,63 @@
+# With Firebase Hosting and Docker example
+
+## How to use
+
+### Using `create-next-app`
+
+Execute [`create-next-app`](https://github.com/segmentio/create-next-app) with [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) or [npx](https://github.com/zkat/npx#readme) to bootstrap the example:
+
+```bash
+npx create-next-app --example with-firebase-hosting-and-docker with-firebase-hosting-and-docker-app
+# or
+yarn create next-app --example with-firebase-hosting-and-docker with-firebase-hosting-and-docker-app
+```
+
+### Download manually
+
+Download the example:
+
+```bash
+curl https://codeload.github.com/zeit/next.js/tar.gz/canary | tar -xz --strip=2 next.js-canary/examples/with-firebase-hosting-and-docker
+cd with-firebase-hosting-and-docker
+```
+
+Set up firebase:
+
+* create a project through the [firebase web console](https://console.firebase.google.com/)
+* grab the projects ID from the web consoles URL: https://console.firebase.google.com/project/<projectId>
+* update the `.env` with your FB_PROJECTID and FIREBASE_TOKEN ( see .env.example )
+* ADD `serviceAccountKey.json` to your project root
+
+### Dev next
+
+```bash
+yarn docker:dev
+open http://localhost:5000
+```
+
+### Firebase serve
+
+```bash
+yarn docker:serve
+open http://localhost:3000
+```
+
+### Firebase deploy
+
+```bash
+yarn docker:deploy
+```
+
+## The idea behind the example
+
+The goal is to host the Next.js app on Firebase Cloud Functions with Firebase Hosting rewrite rules so our app is served from our Firebase Hosting URL, with docker support. Each individual `page` bundle is served in a new call to the Cloud Function which performs the initial server render.
```

Comment:
I think this example should more clearly explain that the use of Docker is entirely for local development and the deployment process and that Firebase itself will not leverage Docker itself.